File: src/main/java/org/goafabric/catalog/job/insurance/InsuranceRepository.java
package org.goafabric.catalog.job.insurance;

import org.springframework.data.jdbc.repository.query.Query;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface InsuranceRepository extends CrudRepository<InsuranceEo, String> {
    @Query("SELECT * FROM catalog.insurance WHERE code = :code")
    List<InsuranceEo> findByCode(String code);

    @Query("SELECT * FROM catalog.insurance WHERE UPPER(display) LIKE CONCAT(UPPER(:display), '%')")
    List<InsuranceEo> findByDisplayStartsWith(String display);
}

File: src/main/kotlin/org/goafabric/catalog/job/insurance/InsuranceRepository.kt
package org.goafabric.catalog.job.insurance

import org.springframework.data.jdbc.repository.query.Query
import org.springframework.data.repository.CrudRepository

interface InsuranceRepository : CrudRepository<InsuranceEo, String> {
    @Query("SELECT * FROM catalog.insurance WHERE code = :code")
    fun findByCode(code: String): MutableList<InsuranceEo>

    @Query("SELECT * FROM catalog.insurance WHERE UPPER(display) LIKE CONCAT(UPPER(:display), '%')")
    fun findByDisplayStartsWith(display: String): MutableList<InsuranceEo>
}
